cmake/TestFunctions.cmake

```diff
@@ -0,0 +1,25 @@
+include_guard(GLOBAL)
+
+# Helper function to create a test which will link against Gtest libraries.
+function(add_complex_test)
+    set(options "")
+    set(oneValueArgs NAME)
+    set(multiValueArgs SOURCES LIBRARIES)
+    cmake_parse_arguments(CUSTOM_TEST "${options}" "${oneValueArgs}"
+                          "${multiValueArgs}" ${ARGN})
+
+    if(NOT DEFINED CUSTOM_TEST_NAME)
+        message(FATAL_ERROR "NAME is required for a complex test.")
+    endif()
+
+    string(CONCAT TEST_TARGET "${CUSTOM_TEST_NAME}" "Target")
+    set(GTEST_LIBS)
```

Comment:
This line might not be needed. List append might create this list variable if it does not exist.

Reply:
Can you try removing it and checking if it works? We might as well remove it here if we dont need it.

Reply:
Removed unneeded statement.